File: testbench/tests/native-server/c/01-hello-server.c
// @probe: GET /
// @expect-status: 200
// @expect-body-contains: Hello from C
#include "nex_http.h"

int main() {
    int port = nex_get_port(8080);
    long long server = nex_server_create(port);

    long long req = nex_server_accept(server);
    long long req_id = nex_req_get(req, "id");

    nex_server_respond(req_id, 200, "Hello from C", "text/plain");
    nex_server_close(server);
    return 0;
}

File: testbench/tests/e2e-deploy/01-hello-server.js
// Basic HTTP server — simplest possible deploy test
// @local-run
// @port: 8090
// @probe: GET /
// @expect-status: 200
// @expect-body-contains: hello from nex
// @probe-count: 3
// @probe-interval: 500

var http = require("http");

var server = http.createServer(function (req, res) {
  res.writeHead(200, { "Content-Type": "text/plain" });
  res.end("hello from nex");
});

server.listen(8090, function () {
  console.log("Server running on port 8090");
});
